transform/transformers/Transformer.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from collections import OrderedDict
+import os.path
+from structlog import wrap_logger
+
+from transform.transformers.CSFormatter import CSFormatter
+from transform.transformers.ImageTransformer import ImageTransformer
+from transform.transformers.survey import Survey
+
+
+class Transformer:
+
+    """A base class for SDX transformers.
+
+    Subclasses should define the contents of the following class variables:
+
+        * :py:const:`defn`
+        * :py:const:`package`
+        * :py:const:`pattern`
+
+    """
+    #: Transformer subclasses declare their transforms in this class variable.
```

Comment:
Maybe have these comments in a docstring instead?

Reply:
these are from an earlier change